photon-core/src/main/java/org/photonvision/common/networking/NetworkManager.java

```diff
@@ -52,102 +52,54 @@ public void initialize(boolean shouldManage) {
-                                        .setStaticCommand
-                                        .replace(NetworkConfig.NM_IFACE_STRING, config.getEscapedInterfaceName())
-                                        .replace(NetworkConfig.NM_IP_STRING, config.staticIp));
-
-                        if (Platform.isRaspberryPi()) {
-                            // Pi's need to manually have their interface adjusted?? and the 5-second sleep is
-                            // integral in my testing (Matt)
-                            shell.executeBashCommand(
-                                    "sh -c 'nmcli con down "
-                                            + config.getEscapedInterfaceName()
-                                            + "; nmcli con up "
-                                            + config.getEscapedInterfaceName()
-                                            + "'");
-                        } else {
-                            // for now just bring down /up -- more testing needed on beelink et al.
-                            shell.executeBashCommand(
-                                    "sh -c 'nmcli con down "
-                                            + config.getEscapedInterfaceName()
-                                            + "; nmcli con up "
-                                            + config.getEscapedInterfaceName()
-                                            + "'");
-                        }
-                    } catch (Exception e) {
-                        logger.error("Error while setting static IP!", e);
-                    }
-                } else {
-                    logger.warn("Got empty static IP?");
-                }
+        if (physicalDevices.stream().noneMatch(it -> (it.devName.equals(config.networkManagerIface)))) {
+            try {
+                var iFace = physicalDevices.stream().findFirst().orElseThrow();
+                logger.warn("The configured interface doesn't match any available interface. Applying configuration to " + iFace.devName);
+                // update NetworkConfig with actual interface
+                config.networkManagerIface = iFace.devName;
+                ConfigManager.getInstance().requestSave();
+            } catch (Exception e) {
+                // already checked that there is at least one item in physicalDevices, so this should never happen
+                logger.error("No valid network interfaces to manage", e);
+                return;
             }
+        }
+
+        logger.info("Setting " + config.connectionType + " with team " + config.ntServerAddress + " on " + config.networkManagerIface);
+
+        // always set hostname (unless it's blank)
+        if (!config.hostname.isBlank()) {
+            setHostname(config.hostname);
         } else {
-            logger.info("Not managing network on non-Linux platforms");
+            logger.warn("Got empty hostname?");
+            // should we force it back to photonvision in this case?
```

Comment:
I think it's fine to just warn -- discovery by mdns is usually secondary anyways, and the UI should protect against this.